File: q2971/q2971.java
class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        long runningSum = nums[0] + nums[1];
        long res = -1;
        for (int i = 2; i < nums.length; i++) {
            if (nums[i] < runningSum) {
                res = nums[i] + runningSum;
            }
            runningSum += nums[i];
        }
        return res;
    }
}

File: q2971/q2971.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        runningSum = nums[0] + nums[1]
        res = -1
        for i in range(2, len(nums)): 
            if runningSum > nums[i]:
                res = runningSum + nums[i]
            runningSum += nums[i]
        return res